manifests/server/grant.pp

```diff
@@ -103,8 +138,9 @@
   }
 
   $_unless = $unless_function ? {
-      false   => undef,
-      default => "SELECT 1 WHERE ${unless_function}('${role}',
+      false    => undef,
+      'custom' => $custom_unless,
+      default  => "SELECT 1 WHERE ${unless_function}('${role}',
```

Comment:
@igalic, `$custom_unless` contains the SQL query that is passed to `Postgresql_psql` (via `$_unless`). It will be executed as is by `Postgresql_psql`, just like the SQL query constructed in the default case here.